src/utils/x11rb.rs

```diff
@@ -36,6 +36,8 @@ use calloop::{
 pub struct X11Source {
     connection: Arc<RustConnection>,
     channel: Option<Channel<Event>>,
+    #[cfg(feature = "xwayland")]
+    pub(crate) sender: SyncSender<Event>,
```

Comment:
What is this sender for? Another internal pub(crate) field isn't exactly very nice for untangling the single monolithic crate in the future.

Reply:
It is used in the `X11Injector`-type to inject events in our processing loop, which is used to ease the wl_surface-id-dance.
This sender already exists inside the source without xwayland, the feature just adds an otherwise unused way to access that sender. So for separating this is not really a problem, we could just make the sender part of a public api of the `X11Source`, if we would split it apart.